src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -865,50 +968,70 @@ private async Task<Dictionary<KeyValueIdentifier, ConfigurationSetting>> LoadKey
             return watchedSettings;
         }
 
-        private Dictionary<KeyValueIdentifier, ConfigurationSetting> UpdateWatchedKeyValueCollections(Dictionary<KeyValueIdentifier, ConfigurationSetting> watchedSettings, IDictionary<string, ConfigurationSetting> existingSettings)
+        private async Task<List<KeyValueChange>> GetRefreshedKeyValueCollections(
+            IEnumerable<KeyValueWatcher> featureFlagWatchers,
+            Dictionary<KeyValueIdentifier, IEnumerable<MatchConditions>> watchedCollections,
+            IDictionary<string, ConfigurationSetting> existingSettings,
+            ConfigurationClient client,
+            CancellationToken cancellationToken)
         {
-            foreach (KeyValueWatcher changeWatcher in _options.MultiKeyWatchers)
+            List<KeyValueChange> keyValueChanges = new List<KeyValueChange>();
+
+            HashSet<string> existingKeys = new HashSet<string>();
+
+            foreach (KeyValueWatcher changeWatcher in featureFlagWatchers)
             {
-                IEnumerable<ConfigurationSetting> currentKeyValues = GetCurrentKeyValueCollection(changeWatcher.Key, changeWatcher.Label, existingSettings.Values);
+                SettingSelector selector = new SettingSelector()
+                {
+                    KeyFilter = changeWatcher.Key,
+                    LabelFilter = changeWatcher.Label ?? LabelFilter.Null
+                };
+
+                await CallWithRequestTracing(async () =>
+                {
+                    AsyncPageable<ConfigurationSetting> pageableSettings = client.GetConfigurationSettingsAsync(selector, cancellationToken);
+
+                    await foreach (Page<ConfigurationSetting> page in _options.PageableManager.GetPages(pageableSettings).ConfigureAwait(false))
+                    {
+                        keyValueChanges.AddRange(page.Values.Select(setting => new KeyValueChange()
+                        {
+                            Key = setting.Key,
+                            Label = setting.Label.NormalizeNull(),
```

Comment:
Why do we need `NormalizeNull` here?

Reply:
yea good point, I don't see why at the moment